decider-HSMetrics/src/main/java/ca/on/oicr/pde/deciders/HSMetricsDecider.java

```diff
@@ -0,0 +1,368 @@
+import net.sourceforge.seqware.common.module.ReturnValue;
+import net.sourceforge.seqware.common.util.Log;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ *
+ * @author prath@oicr.on.ca
+ */
+public class HSMetricsDecider extends OicrDecider {
+
+    private final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.S");
+    private Map<String, BeSmall> fileSwaToSmall;
+
+    private String templateType = null;
+    private String queue = "";
+    private String baseName;
+    private int coverageCap = 500;
+    private float minPCT = (float) 0.5;
+    private String stringency = "LENIENT";
+    
+
+    private final static String BAM_METATYPE = "application/bam";
+    private String tumorType;
+//    private String templateType;
+//    private List<String> duplicates;
+
+    public HSMetricsDecider() {
+        super();
+        fileSwaToSmall = new HashMap<String, BeSmall>();
+        parser.acceptsAll(Arrays.asList("ini-file"), "Optional: the location of the INI file.").withRequiredArg();
+        parser.accepts("template-type", "Required. Set the template type to limit the workflow run "
+                + "so that it runs on data only of this template type").withRequiredArg();
+        parser.accepts("queue", "Optional: Set the queue (Default: not set)").withRequiredArg();
+        parser.accepts("tumor-type", "Optional: Set tumor tissue type to something other than primary tumor (P), i.e. X . Default: Not set (All)").withRequiredArg();
+    }
+
+    @Override
+    public ReturnValue init() {
+        Log.debug("INIT");
+        this.setMetaType(Arrays.asList(BAM_METATYPE));
+        this.setHeadersToGroupBy(Arrays.asList(Header.FILE_SWA));
+
+        ReturnValue rv = super.init();
+        rv.setExitStatus(ReturnValue.SUCCESS);
+
+        //Group by sample if no other grouping selected
+        if (this.options.has("group-by")) {
+            Log.error("group-by parameter passed, but this decider does not allow overriding the default grouping (by Donor + Library Type)");
+        }
+
+        if (this.options.has("queue")) {
+            this.queue = options.valueOf("queue").toString();
+        }
+
+        if (this.options.has("template-type")) {
+            this.templateType = options.valueOf("template-type").toString();
+        } else {
+            if (!options.hasArgument("template-type")) {
+
+                Log.error("--template-type requires an argument, e.g. EX,WT");
+                rv.setExitStatus(ReturnValue.INVALIDARGUMENT);
+                return rv;
+            }
+        }
+
+        return rv;
+    }
+
+    /**
+     * Final check
+     *
+     * @param commaSeparatedFilePaths
+     * @param commaSeparatedParentAccessions
+     *
+     * @return
+     */
+    @Override
+    protected ReturnValue doFinalCheck(String commaSeparatedFilePaths, String commaSeparatedParentAccessions) {
+        String[] filePaths = commaSeparatedFilePaths.split(",");
```

Comment:
Final Check does not make sure that we have one file only, it may create problem with ini file writing